legend-engine-xt-relationalStore-spanner-execution/pom.xml

```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ Copyright 2020 Goldman Sachs
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>legend-engine</artifactId>
+        <groupId>org.finos.legend.engine</groupId>
+        <version>3.16.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>legend-engine-xt-relationalStore-spanner-execution</artifactId>
+    <packaging>jar</packaging>
+    <name>Legend Engine - XT - Relational Store - Spanner - Execution</name>
+
+    <dependencies>
+        <!-- ENGINE -->
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-spanner-protocol</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-executionPlan</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-executionPlan-connection</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-executionPlan-connection-authentication</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-shared-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.finos.legend.engine</groupId>
+            <artifactId>legend-engine-xt-relationalStore-protocol</artifactId>
+        </dependency>
+        <!-- ENGINE -->
+
+        <!-- ECLIPSE COLLECTIONS -->
+        <dependency>
+            <groupId>org.eclipse.collections</groupId>
+            <artifactId>eclipse-collections-api</artifactId>
+        </dependency>
+        <!-- ECLIPSE COLLECTIONS -->
+
+        <!-- Guava -->
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
```

Comment:
From the imports its not clear how this is used. What we need this for that other existing libraries dont do already?

Reply:
Isn't "guava" is "existing" library? It is already added as dependency and spanner-*** projects are new modules (and they might be removed out engine in near future), so I thought that I, as a developer of new module could choose to use this library as long it is not bringing new dependencies.

Reply:
This does not answer my question - why you need guava for?  

As for your comment - We have eclipse as our preferred collection library, and we need to avoid using it guava where our preferred option will suffice.  Also, having a transient dependency is different from having it as compile time, the latter pinning us to a given version and adding to maintenance cost of the module.   For example, if spanner was to remove guava, we will still need to keep it when using other preferred libraries will have allow us to also remove it.

Reply:
> his does not answer my question - why you need guava for?

I used guava for Strings lib here.
https://github.com/roasher/legend-engine/blob/1f935089eba2322ba59a20086f0c65478bfaee82/legend-engine-xt-relationalStore-spanner-execution/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/spanner/driver/SpannerManager.java#L17

so, It is not about collections.
But, I think I can rewrite code without adding this library if it would make sense.

Reply:
You can also consider similar utilities on EC:  https://github.com/eclipse/eclipse-collections/blob/7d74dba3a0bc997d98c1b12c161459fe6fd65bce/eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/StringIterate.java#L1062

Reply:
> You can also consider similar utilities on EC: https://github.com/eclipse/eclipse-collections/blob/7d74dba3a0bc997d98c1b12c161459fe6fd65bce/eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/StringIterate.java#L1062

Ok, got it. Will remove this dependency.
Please, note that I would need to add eclipse-collections instead. So, it is not -1 dependency - it is just other.